### PR TITLE
Jojo

### PR DESCRIPTION
**Pull request by Mahjongasaurus**
I think I set this up right.

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -1,37 +1,110 @@
...
+ for the strength of our robot as a whole.
+ Please don't break.
+
+ We pray to the control systems, and National Instruments,
+ for if we do not we will surely perish
+ as the RoboRio may not work.
+ Please have mercy.
+
+ We pray to the Robot Inspectors and the scale,
+ as even though are robot is small,
```
Comment (RiverDivider) [inline src/main/java/frc/robot/Robot.java]: Grammer yikes 
our*

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -1,37 +1,110 @@
...
-  private RobotContainer m_robotContainer;
 
-  /**
+public class Robot<MyFindTapePipeline> extends TimedRobot {
+  public Command m_autonomousCommand;
+  
+    // A USB camera connected to the roboRIO.
+    public VideoSource usbCamera;
+  public RobotContainer m_robotContainer;
+  public static final IntakeSubsystem intake = new IntakeSubsystem();
```
Comment (RiverDivider) [inline src/main/java/frc/robot/Robot.java]: put this in RobotContainer please

```
  public static final IntakeSubsystem intake = new IntakeSubsystem();
```

File: src/main/java/frc/robot/Robot.java [hunk 3]
```diff
@@ -1,37 +1,110 @@
...
+   /**
    * This function is run when the robot is first started up and should be used for any
    * initialization code.
    */
+  // The pipeline outputs we want
+  public boolean pipelineRan = false; // lets us know when the pipeline has actually run
+  public double angleToTape = 0;
+  public double distanceToTape = 0;
+  
+  public void copyPipelineOutputs(MyFindTapePipeline pipeline) {
```
Comment (RiverDivider) [inline src/main/java/frc/robot/Robot.java:94]: confused as to what 
```
  public void copyPipelineOutputs(MyFindTapePipeline pipeline) {
```
this actually does. clarify please

File: src/main/java/frc/robot/commands/IntakeCommand.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+// Called just before this Command runs the first time
+  @Override
+  public void initialize() {
+  }
+
+  // Called repeatedly when this Command is scheduled to run
+  @Override
+  public void execute() {
```
Comment (RiverDivider) [inline src/main/java/frc/robot/commands/IntakeCommand.java:30]: uhhhh is this just from last year 
our intake this year will only ever spin one way

File: src/main/java/frc/robot/RobotContainer.java [hunk 5]
```diff
@@ -51,13 +51,6 @@ public RobotContainer() {
               -driveController.getRawAxis(1), -driveController.getRawAxis(5)),
               drive)
     );*/
-
-    drive.setDefaultCommand(
-      new RunCommand(
-        () -> drive.testAnglePIDLoop(drive.frontRightModule,
-         (driveController.getRawAxis(0) * .5), (-driveController.getRawAxis(1)) * .5),
-         drive)
-    );
```
Comment (RiverDivider) [inline src/main/java/frc/robot/RobotContainer.java]: go ahead and add in a default command here when you move intake from Robot to RobotContainer
`intake.setDefaultCommand(new IntakeCommand(), intake)`
I believe should work

File: src/main/java/frc/robot/subsystems/IntakeSubsystem.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+  public DigitalInput intakeSwitch = new DigitalInput(kIntakeLimitSwitch);
+  
+  public IntakeSubsystem() {
+    intakeMotor2.setInverted(true);
+  }
+  public void intake() {
+  }
+
+  public void initDefaultCommand() {
```
Comment (RiverDivider) [inline src/main/java/frc/robot/subsystems/IntakeSubsystem.java:32]: Deprecated, see above for where to implement this

File: src/main/java/frc/robot/subsystems/IntakeSubsystem.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+import static frc.robot.Constants.*;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.Spark;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.RobotContainer;
+import frc.robot.commands.IntakeCommand;
+
+public class IntakeSubsystem extends SubsystemBase {
+  // Put methods for controlling this subsystem
```
Comment (RiverDivider) [inline src/main/java/frc/robot/subsystems/IntakeSubsystem.java:18]: if you wanna have this in, it's cool, but port over these PWM motors to CAN motors with the IDs already in the latest version of temp put in

File: src/main/java/frc/robot/commands/IntakeCommand.java (hunk 4 above)
Comment (RiverDivider) [reply src/main/java/frc/robot/commands/IntakeCommand.java:30]: OK I know this probably hasn't been what you were actually working on, it was just in your code.

Talk to me when you see this, I can prob push this up and have someone else/me change the subsystem stuff while you continue with vision stuff.

File: src/main/java/frc/robot/Robot.java [hunk 8]
```diff
@@ -78,10 +153,30 @@ public void autonomousInit() {
 
   /**
    * This function is called periodically during autonomous.
+ * @return 
    */
+  double distanceToWall;
   @Override
   public void autonomousPeriodic() {
-  }
+    double angleCos;
```
Comment (RiverDivider) [inline src/main/java/frc/robot/Robot.java:168]: add docs on purpose of this
Thanks!

(i'm reading it as a constant update to the angle and distance variables, but make that explicitly obvious at the top of the block so other people know exactly what this does)

**Review by RiverDivider — APPROVED** [review]
Vision stuff looks fine, just comment (and I mean I wanna see your comments, not comments already in whatever code you're using as a guide) what everything does so I can read it better.

Most of the issues with this are with the subsystem. I or someone else can fix those easy, I just don't want it in.
(I'll merge this when you read this, just wanted to let you know my plans for changing intake stuff before I do)

File: src/main/java/frc/robot/subsystems/IntakeSubsystem.java (hunk 6 above)
Comment (Mahjongasaurus) [reply src/main/java/frc/robot/subsystems/IntakeSubsystem.java:32]: be more specific as to what "above" is referencing

File: src/main/java/frc/robot/subsystems/IntakeSubsystem.java (hunk 7 above)
Comment (Mahjongasaurus) [reply src/main/java/frc/robot/subsystems/IntakeSubsystem.java:18]: be more specific. What is "this"? the whole block or a specific line?

File: src/main/java/frc/robot/commands/IntakeCommand.java (hunk 4 above)
Comment (Mahjongasaurus) [reply src/main/java/frc/robot/commands/IntakeCommand.java:30]: I see it, and that's fine, it doesn't bother me

File: src/main/java/frc/robot/subsystems/IntakeSubsystem.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  public IntakeSubsystem() {
+  }
+  public void intake() {
+  }
+
+  public void initDefaultCommand() {
+    // Set the default command for a subsystem here.
+    // setDefaultCommand(new MySpecialCommand());
+
+   setDefaultCommand(new IntakeCommand());
```
Comment (RiverDivider) [inline src/main/java/frc/robot/subsystems/IntakeSubsystem.java:36]: this is unnecessary because it is in robotcontainer, remove

**Review by RiverDivider — APPROVED** [review]

**Pull request merged**